### PR TITLE
fixed the errors for zip that now disables the submit button

### PR DESCRIPTION
**Pull request by vikas199**
### Jira Story

- [ If Zipcode is invalid, 'Submit button' in screening card should be disabled INT-1976](https://osi-cwds.atlassian.net/browse/HOT-1976)

## Description
If there are zip errors the submit button should be disabled and there should be no validation message on the read-only address 

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by vikas199** [comment]
jenkins retest this please

File: app/javascript/selectors/screening/personShowSelectors.js [hunk 1]
```diff
@@ -93,11 +93,16 @@ export const getErrorsSelector = (state, personId) => {
...
+  const addressZip =
+  person.get('addresses') === undefined ? [] : person.get('addresses').filter((address) => !address.get('legacy_id')).map((newAddress) => (
+    getZIPErrors(newAddress.get('zip'))
+  )).flatten()
   const roles = person.get('roles', List())
   return fromJS({
     name: getNameErrors(firstName, lastName, roles),
     roles: getRoleErrors(state, personId, roles),
     ssn: getSSNErrors(ssnWithoutHyphens),
+    addressZip: addressZip,
```
[the remark this answers is not in the document]
Comment (vikas199) [reply app/javascript/selectors/screening/personShowSelectors.js]: yes will do it

File: app/javascript/selectors/screening/personShowSelectors.js [hunk 2]
```diff
@@ -93,11 +93,16 @@ export const getErrorsSelector = (state, personId) => {
   const ssnWithoutHyphens = ssn.replace(/-|_/g, '')
   const lastName = person.get('last_name')
   const firstName = person.get('first_name')
+  const addressZip =
+  person.get('addresses') === undefined ? [] : person.get('addresses').filter((address) => !address.get('legacy_id')).map((newAddress) => (
```
[the remark this answers is not in the document]
Comment (NagaPradeepGajula-osi) [reply app/javascript/selectors/screening/personShowSelectors.js]: I would like this to be in multiple line as Will mentioned.

File: spec/javascripts/selectors/screening/personShowSelectorsSpec.js [hunk 3]
```diff
@@ -337,6 +337,26 @@ describe('personShowSelectors', () => {
...
+        expect(getErrorsSelector(state, 'one').get('addressZip'))
+          .toBeImmutable(List([['zip code should be 5 digits']]))
+      })
+      it('should return empty when zip is 5 digits', () => {
+        const people = [{id: 'one', addresses: [{zip: '12345'}]}]
+        const state = fromJS({participants: people})
+        expect(getErrorsSelector(state, 'one').get('addressZip'))
+          .toEqualImmutable(List([]))
+      })
+      it('should return empty when there is no zip', () => {
```
Comment (NagaPradeepGajula-osi) [inline spec/javascripts/selectors/screening/personShowSelectorsSpec.js]: Ok I am confused here. Why would you return empty in both cases 347 and 353? Am I missing something here let me know.

Comment (vikas199) [reply spec/javascripts/selectors/screening/personShowSelectorsSpec.js]: Inline 347 when the zip is 5 digits it should return an empty array without any errors,
Whereas in line 353 if a user enters the complete and do not enter zip then also it should not return an error

**Comment by vikas199** [comment]
jenkins retest this please

**Pull request merged**